src/communication/network_interface_cl.c

```diff
@@ -5682,36 +5736,205 @@ stats_update_all_statistics (int with_fullscan)
+  memset (&query_error, 0, sizeof (DB_QUERY_ERROR));
+  db_make_null (&class_name_val);
 
-  ptr = or_pack_int (request, with_fullscan);
+  error = db_compile_and_execute_local (query, &query_result, &query_error);
+  if (error < 0)
+    {
+      goto end;
+    }
 
-  req_error =
-    net_client_request (NET_SERVER_QST_UPDATE_ALL_STATISTICS, request, OR_ALIGNED_BUF_SIZE (a_request), reply,
-			OR_ALIGNED_BUF_SIZE (a_reply), NULL, 0, NULL, 0);
-  if (!req_error)
+  error = db_query_first_tuple (query_result);
+  if (error != DB_CURSOR_SUCCESS)
     {
-      or_unpack_errcode (reply, &error);
+      goto end;
     }
 
-  return error;
+  do
+    {
+      const char *class_name = NULL;
+
+      /* class_name */
+      error = db_query_get_tuple_value (query_result, 0, &class_name_val);
+      if (error != NO_ERROR)
+	{
+	  goto end;
+	}
+      class_name = db_get_string (&class_name_val);
+
+      class_mop = db_find_class (class_name);
+      if (class_mop == NULL)
+	{
+	  error = ER_FAILED;
+	  goto end;
+	}
+
+      /* get NDV by query */
+      if (stats_get_ndv_by_query (class_mop, &class_attr_ndv, NULL, with_fullscan) != NO_ERROR)
+	{
+	  error = ER_FAILED;
+	  goto end;
+	}
+
+      request_size = OR_INT_SIZE + (sizeof (ATTR_NDV) * (class_attr_ndv.attr_cnt + 1)) + OR_INT_SIZE + OR_OID_SIZE;
+      request = (char *) malloc (request_size);
+      if (request == NULL)
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, (size_t) request_size);
+	  error = ER_OUT_OF_VIRTUAL_MEMORY;
+	  goto end;
+	}
+
+      ptr = or_pack_int (request, class_attr_ndv.attr_cnt);
+      for (int i = 0; i < class_attr_ndv.attr_cnt + 1; i++)
+	{
+	  ptr = or_pack_int (ptr, class_attr_ndv.attr_ndv[i].id);
+	  ptr = or_pack_int64 (ptr, class_attr_ndv.attr_ndv[i].ndv);
+	}
+      ptr = or_pack_oid (ptr, WS_OID (class_mop));
+      ptr = or_pack_int (ptr, with_fullscan);
+
+      reply = OR_ALIGNED_BUF_START (a_reply);
+
+      req_error =
+	net_client_request (NET_SERVER_QST_UPDATE_STATISTICS, request, request_size, reply,
+			    OR_ALIGNED_BUF_SIZE (a_reply), NULL, 0, NULL, 0);
+      if (!req_error)
+	{
+	  or_unpack_errcode (reply, &error);
+	}
+      if (class_attr_ndv.attr_ndv != NULL)
+	{
+	  free_and_init (class_attr_ndv.attr_ndv);
+	}
+      free_and_init (request);
+      db_value_clear (&class_name_val);
+
+      if (error != NO_ERROR)
+	{
+	  goto end;
+	}
+    }
+  while ((error = db_query_next_tuple (query_result)) == DB_CURSOR_SUCCESS);
+
+end:
+  if (query_result)
+    {
+      db_query_end (query_result);
+      query_result = NULL;
+    }
+  if (class_attr_ndv.attr_ndv != NULL)
+    {
+      free_and_init (class_attr_ndv.attr_ndv);
+    }
+  if (request)
+    {
+      free_and_init (request);
+    }
+
+  return (error == DB_CURSOR_END) ? NO_ERROR : error;
+
 #else /* CS_MODE */
-  int success;
+  int error = NO_ERROR;
+  CLASS_ATTR_NDV class_attr_ndv;
+  const char *query = "SELECT class_name FROM db_class";
```

Comment:
I think it should be the same as the query in CS_MODE.

Reply:
I fixed it.